### PR TITLE
Fix many clippy warnings across Kani

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

1. Apply many of clippy's suggested fixes to code where (1) the fix was actually better (2) it was simple.
2. For some I thought were "wrong" I added a couple `#![allow(...` directives.
3. For others, I just left them for now, I think some should be done but require more thinking than I wanted for this PR.

### Testing:

* How is this change tested? existing regression

* Is this a refactor change? yep

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: cprover_bindings/src/goto_program/location.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ impl Location {
                 file.into(),
                 function.intern(),
                 line,
-                col.map(|x| x.try_into().unwrap()),
+                col,
```
Comment (danielsn) [inline cprover_bindings/src/goto_program/location.rs:141]: lol

File: cprover_bindings/src/goto_program/symtab_transformer/gen_c_transformer/name_transformer.rs [hunk 2]
```diff
@@ -48,8 +48,8 @@ impl NameTransformer {
         }
 
         // Don't tranform the `tag-` prefix of identifiers
-        let (prefix, name) = if orig_name.starts_with("tag-") {
-            (&orig_name[..4], &orig_name[4..])
+        let (prefix, name) = if let Some(tag) = orig_name.strip_prefix("tag-") {
+            ("tag-", tag)
```
Comment (danielsn) [inline cprover_bindings/src/goto_program/symtab_transformer/gen_c_transformer/name_transformer.rs:52]: Impressed clippy found this pattern

File: tools/bookrunner/src/util.rs [hunk 3]
```diff
@@ -113,7 +113,7 @@ fn try_parse_args(cur_args: Vec<String>, name: &str, line: &str) -> Vec<String>
     let name = format!("{}-flags:", name);
     let mut split = line.split(&name).skip(1);
     let args: Vec<String> = if let Some(rest) = split.next() {
-        rest.trim().split_whitespace().map(String::from).collect()
+        rest.split_whitespace().map(String::from).collect()
```
Comment (danielsn) [inline tools/bookrunner/src/util.rs:116]: Does this change the semantics?

File: kani-driver/src/call_cbmc.rs [hunk 4]
```diff
@@ -150,15 +150,7 @@ impl KaniSession {
...
-    if let Some(harness_unwind) = args.unwind {
-        Some(harness_unwind)
-    } else if let Some(annotation_unwind) = harness_metadata.unwind_value {
-        Some(annotation_unwind)
-    } else if let Some(default_unwind) = args.default_unwind {
-        Some(default_unwind)
-    } else {
-        None
-    }
+    args.unwind.or(harness_metadata.unwind_value).or(args.default_unwind)
```
Comment (danielsn) [inline kani-driver/src/call_cbmc.rs:153]: This is easier to read :)

File: cprover_bindings/src/irep/to_irep.rs [hunk 5]
```diff
@@ -206,7 +206,7 @@ impl ToIrep for ExprValue {
             }
             //TODO, determine if there is an endineness problem here
             ExprValue::DoubleConstant(i) => {
-                let c: u64 = unsafe { std::mem::transmute(*i) };
+                let c: u64 = i.to_bits();
```
Comment (danielsn) [inline cprover_bindings/src/irep/to_irep.rs:209]: 💯

File: cprover_bindings/src/irep/to_irep.rs [hunk 6]
```diff
@@ -405,8 +405,7 @@ impl ToIrep for StmtBody {
             StmtBody::Assume { cond } => code_irep(IrepId::Assume, vec![cond.to_irep(mm)]),
             StmtBody::AtomicBlock(stmts) => {
                 let mut irep_stmts = vec![code_irep(IrepId::AtomicBegin, vec![])];
-                irep_stmts
-                    .append(&mut stmts.into_iter().map(|x| x.to_irep(mm)).collect::<Vec<Irep>>());
+                irep_stmts.append(&mut stmts.iter().map(|x| x.to_irep(mm)).collect());
```
Comment (danielsn) [inline cprover_bindings/src/irep/to_irep.rs:408]: Bye bye `turbo🐠`

File: cprover_bindings/src/machine_model.rs [hunk 7]
```diff
@@ -46,7 +46,8 @@ impl From<RoundingMode> for BigInt {
     }
 }
 
-/// Constructor
+// TODO: This file should be refactored. This is a bit "OO" style when it doesn't have to be.
```
Comment (danielsn) [inline cprover_bindings/src/machine_model.rs:49]: tracking issue

**Review by danielsn — APPROVED** [review]

File: tools/bookrunner/src/util.rs (hunk 3 above)
Comment (tedinski) [reply tools/bookrunner/src/util.rs:116]: Nope! That's apparently why clippy has this lint. :)

**Pull request merged**